### PR TITLE
Guide update

### PR DESCRIPTION
**Pull request by robertmcnees**
This PR updates the guide in a few main areas:

- Spring Boot Docker Compose Support is the preferred method of running an external Redis container.
- Added a new (common) section on building the guide in 4 different ways. A jar file, a Docker container, a native image, and a native image container
- Replace Lombok with Java Record
- Add GitHub Actions CI build

The teaching content remains unchanged.

### CODE REVIEW

**Comment by robertmcnees** [comment]
Adding @mp911de for an optional review. This is similar in scope to other updates to bring Docker Compose support to our guides.

Pinging @chemicL as an optional reviewer as well.

**Review by chemicL — APPROVED** [review]
From a brief look I think the changes are a great refresh.

File: README.adoc [hunk 1]
```diff
@@ -2,108 +2,62 @@
...
 == What You Will Build
 
 You'll build a Spring application that uses https://projects.spring.io/spring-data-redis/[Spring Data Redis] and https://projectreactor.io/[Project Reactor] to interact with a Redis data store reactively, storing and retrieving `Coffee` objects without blocking. This application uses Reactor's `Publisher` implementations based upon the Reactive Streams specification, namely `Mono` (for a Publisher returning 0 or 1 value) and `Flux` (for a Publisher returning 0 to n values).
 
-== What You Need
+include::https://raw.githubusercontent.com/spring-guides/getting-started-macros/main/guide_introduction.adoc[]
 
-:java_version: 17
-include::https://raw.githubusercontent.com/spring-guides/getting-started-macros/main/prereq_editor_jdk_buildtools.adoc[]
+== Setting up the Redis server
```
Comment (Buzzardo) [inline README.adoc]: Capitalize `server`

File: README.adoc [hunk 2]
```diff
@@ -2,108 +2,62 @@
...
+
+Before you can build a messaging application, you need to set up the server that will
```
Comment (Buzzardo) [inline README.adoc]: Change `that will` to `to`

File: README.adoc [hunk 3]
```diff
@@ -2,108 +2,62 @@
...
+include::https://raw.githubusercontent.com/spring-guides/getting-started-macros/main/docker_compose_support.adoc[]
+
+If you choose to run the Redis server yourself instead of using Spring Boot Docker Compose support, you have a few options:
+- https://redis.io/download[Download the server] and manually run it
+- Install with Homebrew, if you use a Mac
+- Manually run the `compose.yaml` file with `docker compose up`
 
-include::https://raw.githubusercontent.com/spring-guides/getting-started-macros/main/how_to_complete_this_guide.adoc[]
+If you go with any of these alternate approaches, you should remove the `spring-boot-docker-compose` dependency from the Maven or Gradle build file.
+You will also need to add configuration to an `application.properties` file, as described in greater detail in the <<_preparing_to_build_the_application>> section.
```
Comment (Buzzardo) [inline README.adoc]: Remove `will`

File: README.adoc [hunk 4]
```diff
@@ -2,108 +2,62 @@
...
+You can use this https://start.spring.io/#!type=maven-project&language=java&packaging=jar&groupId=com.example&artifactId=demo&name=demo&description=Demo%20project%20for%20Spring%20Boot&packageName=com.example.demo&dependencies=data-redis-reactive,webflux,docker-compose[pre-initialized project] and click Generate to download a ZIP file. This project is configured to fit the examples in this tutorial.
 
 To manually initialize the project:
 
 . Navigate to https://start.spring.io.
 This service pulls in all the dependencies you need for an application and does most of the setup for you.
 . Choose either Gradle or Maven and the language you want to use. This guide assumes that you chose Java.
-. Click *Dependencies* and select *Spring Reactive Web*, *Spring Data Reactive Redis*, and *Lombok*.
+. Click *Dependencies* and select *Spring Reactive Web*, *Spring Data Reactive Redis* and *Docker Compose Support*.
```
Comment (Buzzardo) [inline README.adoc]: Add a comma before the last `and`

File: README.adoc [hunk 5]
```diff
@@ -138,29 +91,83 @@ Create a `RestController` to provide an external interface for our application:
...
+  {
+    "id": "13f13e3a-0798-44b7-8ae4-b319b227bb19",
+    "name": "Darth Redis"
+  }
+]
+```
+
+== Preparing to Build the Application
+
+To run the code without Spring Boot Docker Compose support, you need a version of Redis running locally to connect to.
```
Comment (Buzzardo) [inline README.adoc:130]: Remove `to connect to`

File: README.adoc [hunk 6]
```diff
@@ -138,29 +91,83 @@ Create a `RestController` to provide an external interface for our application:
...
-include::https://raw.githubusercontent.com/spring-guides/getting-started-macros/main/build_an_executable_jar_subhead.adoc[]
+NOTE: No configuration is required in the `application.properties` file because the default values match the Redis server configuration in `compose.yaml`. Specifically, the properties `spring.data.redis.host` and `spring.data.redis.port` default to `localhost` and `6379` respectively.
 
-include::https://raw.githubusercontent.com/spring-guides/getting-started-macros/main/build_an_executable_jar_with_both.adoc[]
+include::https://raw.githubusercontent.com/spring-guides/getting-started-macros/main/build_and_execute_guide.adoc[]
 
+== Test the Application in Docker
 
-== Test the application
+If you ran the application using a Docker instruction above, a simple curl command from a terminal or command line will no longer work.
```
Comment (Buzzardo) [inline README.adoc]: change `above` to `(shown earlier)` - include the parentheses.

File: README.adoc [hunk 7]
```diff
@@ -138,29 +91,83 @@ Create a `RestController` to provide an external interface for our application:
...
+This is because we are running our containers in a https://docs.docker.com/compose/networking/[Docker network] that is not accessible from the terminal or command line. To run curl commands, we can start a third container to run our curl commands and attach it to the same network.
```
Comment (Buzzardo) [inline README.adoc]: Change `are running` to `run`

File: README.adoc [hunk 8]
```diff
@@ -138,29 +91,83 @@ Create a `RestController` to provide an external interface for our application:
...
 
-Now that the application is running, you can test it by accessing `http://localhost:8080/coffees` from HTTPie, curl, or your favorite browser.
+First, obtain an interactive shell to a new container that is running on the same network as the Redis container and the application:
```
Comment (Buzzardo) [inline README.adoc]: Change `is running` to `runs`

**Review by Buzzardo — CHANGES_REQUESTED** [review]

**Pull request merged**